src/dsync/dsync.c

```diff
@@ -3030,16 +3085,52 @@ int main(int argc, char **argv)
         return 1;
     }
 
-    /* create new mfu_file objects */
-    mfu_file_t* mfu_src_file = mfu_file_new();
-    mfu_file_t* mfu_dst_file = mfu_file_new();
-
     /* allocate space for each path */
     mfu_param_path* paths = (mfu_param_path*) MFU_MALLOC((size_t)numargs * sizeof(mfu_param_path));
 
+    /* pointer to path arguments */
+    char** argpaths = &argv[optind];
+
+#ifdef DAOS_SUPPORT
+    /* For error handling */
+    bool daos_do_cleanup = false;
+    bool daos_do_exit = false;
+
+    /* Set up DAOS arguments, containers, dfs, etc. */
+    int daos_rc = daos_setup(rank, argpaths, daos_args, mfu_src_file, mfu_dst_file);
+    if (daos_rc != 0) {
+        daos_do_exit = true;
+    }
+
+    /* DAOS does not support hard links */
+    if (options.link_dest != NULL &&
+            (mfu_src_file->type == DAOS || mfu_src_file->type == DFS) &&
+            (mfu_dst_file->type == DAOS || mfu_dst_file->type == DFS)) {
+        MFU_LOG(MFU_LOG_ERR, "DAOS does not support hard links.");
+        daos_do_cleanup = true;
+        daos_do_exit = true;
+    }
+
+    /* Not yet supported */
+    if (mfu_src_file->type == DAOS || mfu_dst_file->type == DAOS) {
```

Comment:
@daltonbohning if someone is using --daos-api=DAOS even though they pass in two POSIX containers it looks like this will fail. I think that is ok, but the error message might be kind of confusing in that case.

Reply:
@dsikich Hmm... I see what you mean. I updated this.